File: Problem_Solving_CPP/Learning/AS/loop/08_ReverseInterger.cpp
// reverse Integer

#include<iostream>
using namespace std;

int reverse(int num){
    int revNum = 0;
    if (num ==0 ){
        return 0;
    }
    while(num>0 || num<0){
        int rem = num%10;
        revNum = revNum*10 + rem;
        num = num/10;
    }
    return revNum;
}

int main (){
    int num = 123;
    cout<<reverse(num);
}

File: Problem_Solving_CPP/Learning/AS/01_loop/08_ReverseInterger.js
// reverse Integer

function reverseNum (num){
    let sign = 1;
    if (num<0){
        sign = -1;
    }
    
    if (num === 0){
        return 0;
    }
    num = Math.abs(num);
    let revNum = 0;
    while( num>0){
        let rem = num%10;
        revNum = revNum*10 + rem;
        num = Math.floor(num/10);
    }
    return revNum*sign;
}

let num = -123;
let result = reverseNum(num);
console.log(result);


